core/src/lib/address_conf.cc

```diff
@@ -341,6 +339,42 @@ bool SetupPort(unsigned short& port,
   }
 }
 
+bool IsFamilyEnabled(int family)
+{
+  static bool ipv6_enabled = CheckIfFamilyEnabled(AF_INET6);
+  static bool ipv4_enabled = CheckIfFamilyEnabled(AF_INET);
+
+  if (family == AF_INET6) { return ipv6_enabled; }
+  if (family == AF_INET) { return ipv4_enabled; }
+  return false;
+}
```

Comment:
Instead of trying to let the code handle programmer mistakes, we should try to make those mistakes impossible at compile time.  For example here we could do something like this:
```
enum class IpFamily { V4, V6 };

template <IpFamily fam>
bool IsFamilyEnabled() {
  static bool family_enabled[] = {
    CheckIfFamilyEnabled<IpFamily::V4>(),
    CheckIfFamilyEnabled<IpFamily::V6>()
 };
 return family_enabled[static_cast<std::underlying_type_t<IpFamily>>(fam)];
}
```

Reply:
The same is true for the other functions.

Reply:
Alternatively we could also have something like:

```
bool IsFamilyEnabled(IpFamily fam) { ... }
std::optional<IpFamily> GetFamily(int family) { switch (family) {
 case AF_INET: return IpFamily::V4;
 case AF_INET6: return IpFamily::V6;
 default: return std::nullopt; } }
```
So that at least `IsFamilyEnabled` and `CheckIfFamilyEnabled` do not have to check whether the family is a useful value.

Reply:
Alternatively we could have

template <IpFamily fam>
bool IsFamilyEnabled() {
  static bool is_enabled = CheckIfFamilyEnabled<fam>();
  return is_enabled;
}